pkg/persistence/disk_persistence.go

```diff
@@ -65,6 +74,22 @@ func (ds *diskPersistence) getStorageCurrentDirPath() string {
 	return fmt.Sprintf("%s/%s", ds.dataDir, currentDir)
 }
 
+func checkStoragePermission(dirBasePath string) error {
+	_, err := ioutil.ReadDir(dirBasePath)
+	if err != nil {
+		return fmt.Errorf("cannot read from the storage directory: [%v]", err)
+	}
+
+	tempDir, err := ioutil.TempDir(dirBasePath, "temp")
```

Comment:
Maybe a file, I think it's just a little less risky because it's clear there is nothing more than this file. In case of a directory, one may think we'll write some temporary data inside while the client runs.

`ioutil.TempFile(dirBasePath, "write-test.*.tmp")` ?